src/main/java/graphql/execution/conversion/ArgumentConverterEnvironment.java

```diff
@@ -0,0 +1,27 @@
+package graphql.execution.conversion;
+
+import graphql.PublicApi;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLInputType;
+
+/**
+ * Contains the values passed to a {@link graphql.execution.conversion.ArgumentConverter}
+ */
+@PublicApi
+public interface ArgumentConverterEnvironment {
+
+    /**
+     * @return the argument being converted
+     */
+    GraphQLArgument getArgument();
+
+    /**
+     * @return the type of the argument
+     */
+    GraphQLInputType getArgumentType();
+
+    /**
+     * @return the source object to be converted
```

Comment:
I think this is misleading: the source object is the source object for the execution. We should name it differently.

Reply:
done